code/obj/item/wall_trophy.dm

```diff
@@ -0,0 +1,114 @@
+/*
+Contains Wall Trophy code
+
+Subtypes:
+	Fish wall trophy
+*/
+
+
+//wall trophy
+//some things are taken from plate code
+
+/obj/item/wall_trophy
+	name = "Wall Trophy"
+	desc = "Simple wall trophy that can display items on a wall."
+	icon = 'icons/obj/items/fishing_gear.dmi'
+	icon_state = "wall_trophy"
+	item_state = "wall_trophy"
+
+	event_handler_flags = NO_MOUSEDROP_QOL
+
+	//do we have something attached?
+	var/item_added = FALSE
+	//what items are allowed to be attached?
+	var/allowed_item = /obj/item/
+	//can we un anchor the object?
+	var/can_unanchor = TRUE
+	//for spawning it with an item
+	var/initial_item = list()
+	//automatically anchors the item that gets attached to it
+	var/auto_anchor = FALSE
+
+	New()
+		..()
+		/*for mapping
+			checks for item in initial items and adds them to the trophy
+
+		*/
+		BLOCK_SETUP(BLOCK_BOOK)
+		for (var/type in src.initial_item)
+			. = src.add_item(new type(src.loc))
+			if (!.)
+				stack_trace("Failed to add item to trophy with initial items. [identify_object(src)]- likely can not accept this kind of object")
```

Comment:
This should be made not a loop if `initial_item` is a single type instead of a list